Trouble connecting to ADB server on OS X

Hi there!

I'm currently looking into porting my codebase (C# .NET Framework 4.5.2 with VS2015) to OS X (and further Linux) and I'm running into an issue while connecting to the ADB server using `AdbClient.Instance.Connect(hostname);` where hostname is a local network IP á la 192.168.0.x.

The error returned is

``` C#
[ERROR] FATAL UNHANDLED EXCEPTION: System.Net.Sockets.SocketException: Connection refused
  at System.Net.Sockets.Socket.Connect (System.Net.EndPoint remoteEP) <0x86a98c0 + 0x00167> in <filename unknown>:0 
  at SharpAdbClient.TcpSocket.Connect (System.Net.EndPoint endPoint) <0x86a9868 + 0x0001f> in <filename unknown>:0 
  at SharpAdbClient.AdbSocket..ctor (System.Net.EndPoint endPoint) <0x86a75c8 + 0x0007c> in <filename unknown>:0 
  at SharpAdbClient.Factories+<>c.<Reset>b__17_0 (System.Net.EndPoint endPoint) <0x86a7580 + 0x0002f> in <filename unknown>:0 
  at SharpAdbClient.AdbClient.Connect (System.Net.DnsEndPoint endpoint) <0x86b4570 + 0x00039> in <filename unknown>:0 
  at SharpAdbClient.AdbClientExtensions.Connect (IAdbClient client, System.String host) <0x86b4390 + 0x00076> in <filename unknown>:0 
  at CoCB.Utils.AdbHelper.ConnectToHost (System.String hostname) <0x86b4330 + 0x0001b> in <filename unknown>:0 
  at CoCB.Utils.AdbHelper.Connect () <0x86b4298 + 0x0006b> in <filename unknown>:0 
  at CoCB.Program.Main () <0x6bdf38 + 0x0012b> in <filename unknown>:0 
```

The same code is confirmed to work on Windows 10. I can see that under Unix based operating systems, you take a different approach to connecting to the ADB server, which connects to a socket at "/tmp/5037". I'm not very proficient with *nix sockets, so I don't know if I'm doing something wrong here. I can't find the requested socket file in the file system (through Finder or ls).

The server is started through `AdbServer.StartServer("/Users/xxx/Library/Android/sdk/platform-tools/adb", true);` and returns code 'Started' so I'm assuming the server was started successfully. But somehow the connection still fails. If I try to open a connection with 

``` C#
var socket = new Socket (AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.IP);

socket.Connect ("192.168.0.106", 5555);
```

the connection opens just fine (socket.Connected is True).

Would you have any idea where I could start with this?

Cheers
Philipp

**EDIT**

Replacing TcpSocket.cs

``` C#
//this.socket = new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.IP);
this.socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
```

and AdbServer.cs

``` C#
//EndPoint = new UnixEndPoint($"/tmp/{AdbServerPort}");
EndPoint = new IPEndPoint(IPAddress.Loopback, AdbServerPort);
```

seems to connect to the device. I can query the DeviceData class, but I haven't tried sending any commands yet. Is there any specific reason you are using Mono.Posix specific endpoints for OS X and Linux?

--- a/SharpAdbClient/AdbServer.cs
+++ b/SharpAdbClient/AdbServer.cs
@@ -84,11 +84,12 @@ namespace SharpAdbClient
             switch (Environment.OSVersion.Platform)
             {
                 case PlatformID.Win32NT:
+                case PlatformID.MacOSX:
                     this.EndPoint = new IPEndPoint(IPAddress.Loopback, AdbServerPort);
                     break;
 
-                case PlatformID.MacOSX:
                 case PlatformID.Unix:
+                    // TODO: Only use Unix end points on Debian & derivatives.
                     this.EndPoint = new UnixEndPoint($"/tmp/{AdbServerPort}");
                     break;
 

--- a/SharpAdbClient/TcpSocket.cs
+++ b/SharpAdbClient/TcpSocket.cs
@@ -119,10 +119,11 @@ namespace SharpAdbClient
             switch (platform)
             {
                 case PlatformID.Win32NT:
+                case PlatformID.MacOSX:
                     return new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
 
                 case PlatformID.Unix:
-                case PlatformID.MacOSX:
+                    // TODO: Only return Unix sockets on Debian & derivatives.
                     return new Socket(AddressFamily.Unix, SocketType.Stream, ProtocolType.IP);
 
                 default:

